lib/src/simple_tokenizer.dart

```diff
@@ -0,0 +1,200 @@
+    }
+    if (_scanner.scan(_allElementMatches)) {
+      Match match = _scanner.lastMatch;
+      if (matchesGroup(match, 1)) {
+        return new NgSimpleToken.closeBracket(offset);
+      }
+      if (matchesGroup(match, 2)) {
+        return new NgSimpleToken.bang(offset);
+      }
+      if (matchesGroup(match, 3)) {
+        return new NgSimpleToken.dash(offset);
+      }
+      if (matchesGroup(match, 4)) {
+        return new NgSimpleToken.closeParen(offset);
+      }
+      if (matchesGroup(match, 5)) {
+        _state = _NgSimpleScannerState.text;
+        return new NgSimpleToken.tagEnd(offset);
+      }
+      if (matchesGroup(match, 6)) {
+        return new NgSimpleToken.forwardSlash(offset);
+      }
+      if (matchesGroup(match, 7)) {
+        return new NgSimpleToken.openBracket(offset);
+      }
+      if (matchesGroup(match, 8)) {
+        return new NgSimpleToken.openParen(offset);
+      }
+      if (matchesGroup(match, 9)) {
+        return new NgSimpleToken.whitespace(offset, _scanner.substring(offset));
+      }
+      if (matchesGroup(match, 10)) {
+        return new NgSimpleToken.text(offset, _scanner.substring(offset));
+      }
+      if (matchesGroup(match, 11)) {
+        return new NgSimpleToken.text(offset, _scanner.substring(offset));
+      }
+      if (matchesGroup(match, 12)) {
+        return new NgSimpleToken.doubleQuotedText(
+            offset, _scanner.substring(offset));
+      }
+      if (matchesGroup(match, 14)) {
+        return new NgSimpleToken.singleQuotedText(
+            offset, _scanner.substring(offset));
+      }
+      if (matchesGroup(match, 16)) {
+        return new NgSimpleToken.doubleQuote(offset);
+      }
+      if (matchesGroup(match, 17)) {
+        return new NgSimpleToken.singleQuote(offset);
+      }
+      if (matchesGroup(match, 18)) {
+        return new NgSimpleToken.tagStart(offset);
+      }
+      if (matchesGroup(match, 19)) {
+        return new NgSimpleToken.equalSign(offset);
+      }
+      if (matchesGroup(match, 20)) {
+        return new NgSimpleToken.star(offset);
+      }
+    }
+    return new NgSimpleToken.unexpectedChar(
+        offset, new String.fromCharCode(_scanner.readChar()));
+  }
+
+  NgSimpleToken scanText() {
+    int offset = _scanner.position;
+    if (_scanner.peekChar() == null || _scanner.rest.length == 0) {
+      return new NgSimpleToken.EOF(offset);
+    }
+    if (_scanner.scan(_allTextMatches)) {
+      Match match = _scanner.lastMatch;
+      if (matchesGroup(match, 1)) {
+        return new NgSimpleToken.text(offset, _scanner.substring(offset));
+      }
+      if (matchesGroup(match, 2)) {
+        _state = _NgSimpleScannerState.comment;
+        return new NgSimpleToken.commentBegin(offset);
+      }
+      if (matchesGroup(match, 3)) {
+        _state = _NgSimpleScannerState.element;
+        return new NgSimpleToken.tagStart(offset);
+      }
+    }
+    return new NgSimpleToken.unexpectedChar(
```

Comment:
Might be better to report the error internally and move past it rather than return it